src/Modal/ModalContents.js

```diff
@@ -218,20 +218,16 @@ class ModalContents extends Util.mixin(BindMixin, KeyDownMixin) {
 
   getCloseButton() {
     let props = this.props;
-    if (!props.showCloseButton) {
-      return null;
+
+    if (props.closeButton) {
+      return (
+        <button className={props.closeButtonClass} onClick={this.closeModal}>
```

Comment:
Talked with @jfurrow offline, and decided to make this completely free of boundaries, so it doesn't even wrap it in a button. That way you could just pass:
```js
closeButton={<button onClick={() => { this.setState({open: false}); }}>Close</button>}
```